### PR TITLE
Type cleanup & standardization work

### LINKED ISSUE #2806: Update existing heading/paragraph styles to standardized type
By referring to the brand standards typography issue: https://github.com/MozillaFoundation/Design/issues/319, let's clean up and update current type on the site.

This focuses on heading, copy/paragraph and introducing a dark theme. Any typography in UI components & other elements (form, nav etc.) will be dealt with in different issues.

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #2806 

- We'll need to have a follow-up to clean up font related CSS rules in individual `.scss` files. I tried to do it as much as I could but there are just too many old, unused files sitting in the codebase which made my cleanup work challenging. Once this PR is merged and those unused files are removed, we should have a follow-up PR to finish the CSS rules cleanup.
- Buyer's Guide styling still need to be cleaned up (outside of the scope of this ticket/PR)

--

# Reminder

Since new stylings and changes have been introduced in the new heading hierarchy system (e.g., `<h1>` `<h2>` `<h3>` `<h4>` `<h5>` now have new stylings), **right after** we push this PR to production we'll have to go through CMS pages and make changes to heading type to make sure headings are in the right style we want. e.g., change `<h3>` elements in CMS to `<h2>`

**We need to make sure we have time blocked for updating things in CMS pages before we push this live**

---

**Changes made to h1-h4**

1. old `h1` default stylings have been updated
2. old `h2` styling set becomes the new `h4` styling set
3. old `h3` styling set becomes the new `h2` styling set
4. old `h4` styling set becomes the new `h3` styling set 

For details see https://github.com/MozillaFoundation/Design/issues/319#issuecomment-469926270

### CODE REVIEW

**Comment by mmmavis** [comment]
@sabrinang this is ready for design review. Let's go over the changes together in office tomorrow!

**Comment by mmmavis** [comment]
I'll assign someone for code review once this passes design review.

File: source/sass/mixins.scss [hunk 1]
```diff
@@ -1,3 +1,9 @@
+@mixin heading($font-family, $font-weight, $letter-spacing) {
```
Comment (mmmavis) [inline source/sass/mixins.scss]: This `heading` mixin makes sure all headings will have all three rules set to avoid unintentional styling inheritance.

**Comment by mmmavis** [comment]
@Pomax @youriwims 

This is a HUGE PR with a ton of tedious changes so I don't expect you both to go through every single code changes. I think it would be enough if you can just focus on changes made to `.scss` files as @sabrinang and I have already gone through few rounds of peer design-code review.

If you wanna learn more about the ongoing brand work, @sabrinang has things documented here https://foundation.mozilla.org/en/docs/brand/design/Typography/

**Comment by mmmavis** [comment]
Oh the typography section on [style guide] has been updated too (in case you'd like to an overview list )

File: source/js/donate-modal/donate-modal.jsx [hunk 2]
```diff
@@ -94,8 +94,8 @@ class DonateModal extends React.Component {
         <div className="container">
           <div className="row align-items-center text-center text-md-left">
             <div className="col-md-6">
-              <h1 className="h3-heading">{title}</h1>
-              <p className="normal">{subheading} </p>
```
[the remark this answers is not in the document]
Comment (mmmavis) [reply source/js/donate-modal/donate-modal.jsx]: Yea `.normal` purposely removed here. If you look at https://github.com/mozilla/foundation.mozilla.org/pull/2810/files#diff-736d8c9639437c97fab98e0f42b93814R151 `<p>` without classname will have the same styling set as `.body`. Here we are using `<p>` so adding `.body` would be redundant!

[the remark this answers is not in the document]
Comment (youriwims) [reply source/js/donate-modal/donate-modal.jsx]: Got ittttttt—thanks!

File: source/sass/colors.scss [hunk 3]
```diff
@@ -29,5 +29,6 @@ $white: #ffffff;
 $red: #ff4f5e;
 $dark-red: #cc0011;
 
+$light-blue: #b7b9fa;
```
[the remark this answers is not in the document]
Comment (sabrinang) [reply source/sass/colors.scss]: Thanks for pointing this out! I updated my original comment and yes we should be using `#b7b9fa`

[the remark this answers is not in the document]
Comment (youriwims) [reply source/sass/colors.scss]: Awesome 💃  !

File: source/js/donate-modal/donate-modal.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmmavis) [reply source/js/donate-modal/donate-modal.jsx]: nice catch though!

[the remark this answers is not in the document]
Comment (youriwims) [reply source/js/donate-modal/donate-modal.jsx]: Should've taken a break in between look at this because my eyes were like 👀  💥  💀 lol

**Pull request merged**